src/libcpu/src/jit/jit_integer.cpp

```diff
@@ -601,15 +601,24 @@ neg(PPCEmuAssembler& a, Instruction instr)
 {
    auto eaxLockout = a.lockRegister(asmjit::x86::rax);
 
-   auto dst = a.loadRegisterWrite(a.gpr[instr.rD]);
```

Comment:
Please follow the existing pattern of using allocGpTmp(src) to avoid collisions rather than writing out all the if statements.